### PR TITLE
RHCLOUD-28334 | feature: email processor tests

### PR DESCRIPTION
**Pull request by MikelAlejoBR**
The goal is to have the email processor fully tested.

## Depends on
* https://github.com/RedHatInsights/notifications-backend/pull/2163
* https://github.com/RedHatInsights/notifications-backend/pull/2164

## Links
[[RHCLOUD-28334]](https://issues.redhat.com/browse/RHCLOUD-28334)

### CODE REVIEW

**Comment by gwenneg** [comment]
Could you please fix the build error?
```
Build step io.quarkus.arc.deployment.ArcProcessor#generateResources threw an exception: jakarta.enterprise.inject.spi.DeploymentException: java.lang.IllegalStateException: @javax.inject.Inject declared on com.redhat.cloud.notifications.processors.email.EmailProcessorTest.emailProcessor, use @jakarta.inject.Inject instead
```

File: engine/src/test/java/com/redhat/cloud/notifications/processors/email/EmailProcessorTest.java [hunk 1]
```diff
@@ -0,0 +1,400 @@
...
+        // Verify that the compilation functions were called.
+        Mockito.verify(this.templateService, Mockito.times(1)).compileTemplate(subjectData, "subject");
+        Mockito.verify(this.templateService, Mockito.times(1)).compileTemplate(bodyData, "body");
+
+        // Verify that the rendering functions were called.
+        Mockito.verify(this.templateService, Mockito.times(1)).renderTemplate(event.getEventWrapper().getEvent(), subjectTemplateInstance);
+        Mockito.verify(this.templateService, Mockito.times(1)).renderTemplate(event.getEventWrapper().getEvent(), bodyTemplateInstance);
+
+        // Verify that, since there were no subscribers to notify, the event
+        // was ignored.
```
Comment (gwenneg) [inline engine/src/test/java/com/redhat/cloud/notifications/processors/email/EmailProcessorTest.java:292]: I probably approved something wrong in `EmailProcessor`. If the users preferences are ignored, we should send the notification to the email connector when there are no subscribers.

Comment (MikelAlejoBR) [reply engine/src/test/java/com/redhat/cloud/notifications/processors/email/EmailProcessorTest.java:292]: I will open a PR then to fix that: if we don't find any subscribers in the database, we still keep going with the execution, then:

https://github.com/RedHatInsights/notifications-backend/blob/e65f278eeb83d0aada5c00907c70405ee38f26af/engine/src/main/java/com/redhat/cloud/notifications/processors/email/EmailProcessor.java#L67-L73

Comment (gwenneg) [reply engine/src/test/java/com/redhat/cloud/notifications/processors/email/EmailProcessorTest.java:292]: As discussed in Slack, we'll deal with that in a subsequent PR.

**Review by gwenneg — APPROVED** [review]
LGTM

**Pull request merged**